ui/prometheus-plugin/src/plugins/prometheus-datasource.tsx

```diff
@@ -11,25 +11,17 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { BuiltinVariableDefinition, DurationString, RequestHeaders } from '@perses-dev/core';
+import { BuiltinVariableDefinition } from '@perses-dev/core';
 import { DatasourcePlugin } from '@perses-dev/plugin-system';
 import { instantQuery, rangeQuery, labelNames, labelValues, PrometheusClient } from '../model';
-import {
-  PrometheusDatasourceEditor,
-  PrometheusDatasourceSpec as PrometheusDatasourceSpecFull,
-} from './PrometheusDatasourceEditor';
-
-export interface PrometheusDatasourceSpec {
-  directUrl?: string;
-  headers?: RequestHeaders;
-  scrapeInterval?: DurationString;
-}
+import { PrometheusDatasourceSpec } from './types';
+import { PrometheusDatasourceEditor } from './PrometheusDatasourceEditor';
 
 /**
  * Creates a PrometheusClient for a specific datasource spec.
  */
 const createClient: DatasourcePlugin<PrometheusDatasourceSpec, PrometheusClient>['createClient'] = (spec, options) => {
-  const { directUrl, headers: specHeaders } = spec;
+  const { directUrl, proxy } = spec;
   const { proxyUrl } = options;
```

Comment:
FYI `proxyUrl` is the "proxified" url ready to be used for the query calls (see buildProxyUrl func), it is not equal to `proxy.spec.url` here, hence why this part is not touched.